systers_portal/community/views.py

```diff
@@ -518,3 +520,21 @@ class CommunitySearch(ListView):
     """Search for Communities View"""
     template_name = "community/community_search.html"
     model = Community
+
+    def get(self, request):
+        if request.method == 'GET':
+            context = {}
+            url_parameter = request.GET.get("query")
+            if url_parameter:
+                communities = Community.objects.filter(name__icontains=url_parameter)
+            else:
+                communities = Community.objects.all()
+            context['communities'] = communities
+            if request.is_ajax():
```

Comment:
Did you not update this file? IIRC we decided to not have this code at all

Reply:
I think that is exactly what he has mentioned above. That he was able to add tests for the current execution.

Reply:
I see. I really like @sammy1997's idea though. What do you think?

Reply:
@SanketDG I had started working on the way Sammy suggested but the code started getting more and more complex and unreadable. And after maps would be added it would become even more complex. Also after trying out a few stuff I found that the tests can be done as the context object contains the search result
```python
return render(request, "community/community_search.html", context=context)
```
So the tests can be done by simply testing the context data. Below is an example of how messy things could get. Also on working further the links could not be accesed like this. 
```html
<a href="{%url 'view_community_landing' community.slug %}">
```
So I decided to stick with this. 
![Screenshot from 2020-06-09 21-04-56](https://user-images.githubusercontent.com/43514828/84168356-f1fd1c00-aa94-11ea-83ca-d06f6a157e47.png)

Reply:
@ritwickraj78 haven't you pushed the tests? Also, can you add GIFs of the working feature in the PR body whenever you push a new PR? It helps keep track of the intended feature as not all members will build the code to check it. You can do a short screen recording of the search and make it a GIF and add it to the PR description

Reply:
@sammy1997 I haven't pushed the tests yet. Thought of pushing them after the feature is complete. Also I have already added gif for all the PRs which require one

Reply:
@ritwickraj78 As a last effort, could you see if https://stackoverflow.com/a/11782407 fits your needs?

Reply:
@SanketDG Isn't this the same things I have already done? Overriding the JsonResponse method when the method is ajax.